net/libfab/libfab.c

```diff
@@ -722,24 +527,42 @@ static void libfab_tm_buf_done(struct m0_fab__tm *ftm)
 /**
  * Constructs an address in string format from the connection data parameters
  */
-static void libfab_straddr_gen(struct m0_fab__conn_data *cd, char *buf,
-			       uint8_t len, struct m0_fab__ep_name *en)
-{
+/** TODO: Replace this function with m0_net_ip_print() */
+static void libfab_straddr_gen(struct m0_fab__conn_data *cd,
+			       struct m0_fab__ep_name *en)
+{
+	struct m0_net_ip_addr *addr = &cd->fcd_addr;
+
+	if (addr->na_format == M0_NET_IP_LNET_FORMAT)
```

Comment:
Move common code to a a function to avoid duplication.

Reply:
Done